pkg/mountinfo/mountinfo.go

```diff
@@ -129,35 +132,40 @@ func GetMountInfo() ([]*MountInfo, error) {
 	return parseMountInfoFile(fMounts)
 }
 
-func isMountFS(mountInfos []*MountInfo, mntType string, mapRoot string) (bool, bool) {
-	var mapRootMountInfo *MountInfo
-
-	for _, mountInfo := range mountInfos {
-		if mountInfo.MountPoint == mapRoot {
-			mapRootMountInfo = mountInfo
-			break
-		}
+// IsMountFS returns two boolean values, checking
+// - whether the mapRoot is a mount point;
+// - whether that filesystem type is mntType.
+// Note that
+// - mapRoot should be an absolute and cleaned path;
+// - mapRoot should not be '/'.
+func IsMountFS(mntType int64, mapRoot string) (bool, bool, error) {
```

Comment:
We should probably also figure out a better way to add a unit test or runtime test for this function apart from the existing `(*MountInfoTestSuite).TestIsMountFS`. That one currently only checks whether mountinfo can be parsed correctly.

Reply:
Hmm, again I'm not sure what you mean. The test case was modified to actually check for a mount (assuming that `/proc` always exist, which I believe should be the case even in weird test environments).

I can also add a test case which does actual mount (e.g. tmpfs); will require root though.

Reply:
added